### ISSUE TITLE
Error when converting from legacy to mesh

### ISSUE THREAD

**Issue by arpan-kusari** [isl-org/Open3D#4802]
This issue was closed with the following command (https://github.com/isl-org/Open3D/issues/4800) but I still get an error. I have tried 

```python
t_mesh = o3d.t.geometry.TriangleMesh.from_legacy(mesh)
```

When I run this command, I get this error: 
```

Traceback (most recent call last):
  File "C:\Users\kusari\Documents\Python Projects\Topology-preserving-registration\main.py", line 116, in <module>
    mesh = o3d.t.geometry.TriangleMesh.from_legacy(mesh)
TypeError: from_legacy(): incompatible function arguments. The following argument types are supported:
    1. (mesh_legacy: open3d.cpu.pybind.geometry.TriangleMesh, vertex_dtype: open3d.cpu.pybind.core.Dtype = Float32, triangle_dtype: open3d.cpu.pybind.core.Dtype = Int64, device: open3d.cpu.pybind.core.Device = CPU:0) -> open3d.cpu.pybind.t.geometry.TriangleMesh

```

_Originally posted by @yxlao in https://github.com/isl-org/Open3D/issues/4800#issuecomment-1049654957_

**Comment by theNded** [comment]
When you file a issue you need to post your entire snippet and/or the full error log. At least you need to include the line 'invoked by ...'

**Comment by arpan-kusari** [comment]
Sorry Wei. I have the pcd file of the Stanford Bunny and I want to go from the pcd file to generate mesh and then do raycasting on the mesh

```python
bunny = o3d.io.read_point_cloud('stanford_bunny_0deg.pcd')
bunny.estimate_normals()
# to obtain a consistent normal orientation
bunny.orient_normals_towards_camera_location(bunny.get_center())
mesh = o3d.geometry.TriangleMesh.create_from_point_cloud_poisson(bunny, depth=9)
scene = o3d.t.geometry.RaycastingScene()
scene.add_triangles(mesh)
```
I get the error:

```
TypeError: add_triangles(): incompatible function arguments. The following argument types are supported:
1. (self: open3d.cpu.pybind.t.geometry.RaycastingScene, vertex_positions: open3d.cpu.pybind.core.Tensor, triangle_indices: open3d.cpu.pybind.core.Tensor) -> int
2. (self: open3d.cpu.pybind.t.geometry.RaycastingScene, mesh: open3d.cpu.pybind.t.geometry.TriangleMesh) -> int
```

@yxlao suggested using the following command to convert from legacy mesh (https://github.com/isl-org/Open3D/issues/4800): 

```python
mesh = o3d.t.geometry.TriangleMesh.from_legacy(mesh)
```
but that yields the error: 

```
Traceback (most recent call last):
  File "C:\Users\kusari\Documents\Python Projects\Topology-preserving-registration\main.py", line 116, in <module>
    mesh = o3d.t.geometry.TriangleMesh.from_legacy(mesh)
TypeError: from_legacy(): incompatible function arguments. The following argument types are supported:
    1. (mesh_legacy: open3d.cpu.pybind.geometry.TriangleMesh, vertex_dtype: open3d.cpu.pybind.core.Dtype = Float32, triangle_dtype: open3d.cpu.pybind.core.Dtype = Int64, device: open3d.cpu.pybind.core.Device = CPU:0) -> open3d.cpu.pybind.t.geometry.TriangleMesh
```

**Comment by theNded** [comment]
As I have mentioned, it is more helpful if you post the entire modified snippet and the full error log. We need to see the line "Invoked by ..." to know what was wrong with the argument.

**Comment by arpan-kusari** [comment]
Hi Wei, pasting part of the line - 

```
Invoked with: (TriangleMesh with 93190 points and 186601 triangles., DoubleVector[3, 3.03317, 3.05018, 3.05379, 3.09684, 3, 3.02723, 2.86706, 3.00492,
```

**Comment by theNded** [comment]
Let me rephrase one more time:

Please paste the full code snippet to reproduce the error, and the full error log.

**Comment by arpan-kusari** [comment]
Full code - 
```python
bunny_0 = o3d.io.read_point_cloud('stanford_bunny_0deg.pcd')
    bunny_0.estimate_normals()
    # to obtain a consistent normal orientation
    bunny_0.orient_normals_towards_camera_location(bunny_0.get_center())
    mesh = o3d.geometry.TriangleMesh.create_from_point_cloud_poisson(bunny_0, depth=9)
    mesh = o3d.t.geometry.TriangleMesh.from_legacy(mesh)
    o3d.visualization.draw_geometries([mesh])
    scene = o3d.t.geometry.RaycastingScene()
    scene.add_triangles(mesh)
```
The error log is really long and exceeds the limit. So I have pasted it over here - https://justpaste.it/9lqmp

**Comment by theNded** [comment]
Check out the documents: http://www.open3d.org/docs/latest/tutorial/Advanced/surface_reconstruction.html
`create_from_point_cloud_poisson` returns a **tuple** of mesh and a density vector. `from_legacy` is trying to convert from a tuple.

You will need to change the line to:
`mesh, _ = o3d.geometry.TriangleMesh.create_from_point_cloud_poisson(bunny_0, depth=9)`

Since you are using the tensor API, you will need to use 
`o3d.visualization.draw([mesh])`
in the following step.

**Comment by arpan-kusari** [comment]
Thank you for your help, Wei.
